state/protocol/protocol_state/protocol_statemachine.go

```diff
@@ -42,6 +44,15 @@ func newStateMachine(view uint64, parentState *flow.RichProtocolStateEntry) (*pr
 	}, nil
 }
 
+// NewStateMachine instantiates a new `protocolStateMachine` and returns is as an interface type `ProtocolStateMachine`.
+// Thereby, `NewStateMachine` is a specific version of `StateMachineFactoryMethod` for creating a happy-path state machine.
+//
+// An exception is returned in case the `InvalidEpochTransitionAttempted` flag is set in the `parentState`. This means that
+// the protocol state evolution has reached an undefined state from the perspective of the happy path state machine.
+func NewStateMachine(candidateView uint64, parentState *flow.RichProtocolStateEntry) (ProtocolStateMachine, error) {
+	return newStateMachine(candidateView, parentState)
+}
```

Comment:
I have a slight preference to keep it private unless we need it public, not everytime but in this case it seems viable since we are dealing with factory.

Reply:
+1 Happy to keep it private as well, but it should be one or the other, not both 😅